qiskit/optimization/problems/quadratic_program.py

```diff
@@ -152,36 +153,139 @@ def _add_variable(self,
                       upperbound: Union[float, int] = INFINITY,
                       vartype: VarType = VarType.CONTINUOUS,
                       name: Optional[str] = None) -> Variable:
-        """Checks whether a variable name is already taken and adds the variable to list and index
-        if not.
+        return self._add_variables(lowerbound, upperbound, vartype, name)[1][0]
+
+    def _add_variables(self,
+                       lowerbound: Union[float, int] = 0,
+                       upperbound: Union[float, int] = INFINITY,
+                       vartype: VarType = VarType.CONTINUOUS,
+                       name: Optional[str] = None,
+                       key_format: Optional[str] = None,
+                       keys: Union[int, Sequence] = 1) -> Tuple[List[str], List[Variable]]:
+        if isinstance(keys, int) and keys < 1:
+            raise QiskitOptimizationError(
+                "Cannot create non-positive number of variables: {}".format(keys))
+        if not name and key_format:
+            name = 'x'
+        if name and not key_format:
+            if isinstance(keys, int) and keys > 1 or isinstance(keys, Sequence) and len(keys) > 0:
+                key_format = '{}'
```

Comment:
There are four conditions here without any brackets, I'd add a text comment what is expected to have here for further editors of the code.

Reply:
I think the expected behavior is apparent in the code. The comment I would write is basically how the code is read. But if @stefan-woerner and @t-imamichi insist, I can try to rephrase it.